### PR TITLE
Re-add `http.target` to Django old sem conv server duration metric

### PR DESCRIPTION
**Pull request by lzchen**
Fixes https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2745

https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2714 introduced a regression outlined in the issue above. Going forward, we will probably try to discourage additional contributions that pertain to old HTTP semantic conventions to avoid this kind of mixup happening. We do not want to be supporting multiple "old" versions of the semantic conventions.

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py
Comment (alexmojaki) [inline instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py:813]: I really think this should be more concrete to prevent other similar mistakes slipping through, with something like this:

https://github.com/open-telemetry/opentelemetry-python-contrib/blob/c45a6201e234f73322fcbcb858e906a55d349796/instrumentation/opentelemetry-instrumentation-fastapi/tests/test_fastapi_instrumentation.py#L577-L602

File: instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py
Comment (alexmojaki) [inline instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py:453]:
```suggestion
        target = req_attrs.get(SpanAttributes.HTTP_ROUTE)
```

`target` in metrics ~ `route` in spans

File: instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py
Comment (lzchen) [reply instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py:813]: I agree. We probably would want to rewrite all instrumentation tests to better reflect the attributes actually expected instead of simply comparing with a static list. At least for this PR, it would be better to fix this regression separately from the larger effort of rewriting these tests and to have a separate PR to address that.

File: instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py
Comment (lzchen) [reply instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py:453]: Not entirely sure what you mean by this? We parsing duration attributes so this pertains to metrics.

File: instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py
Comment (alexmojaki) [reply instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py:813]: It's genuinely hard right now to tell if this PR (or past versions of it) are doing the correct thing.

File: instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py
Comment (alexmojaki) [reply instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py]: I think `req_attrs` had a bunch of span attributes as well. Looking more closely I think this was probably fine, and the change you just made feels better still. I'd be more confident that this won't break in future if (1) the tests were stricter and (2) span attributes already had `http.target` as they should.

File: instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py
Comment (lzchen) [reply instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py]: I've added some stricter tests to assert the actual values of expected attributes. Please take a look if they fall along the lines of what you'd expect.

File: instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py
Comment (lzchen) [reply instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py]: I've added stricter tests that assert values and keys of the attributes instead of just the presence of keys that follow a static list. In regards to (2), I believe span attributes are populated by [wsgi](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py#L210) middleware and `http.target` is already [populated](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py#L348).

Comment (alexmojaki) [reply instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py]: Thanks, I do think that's an improvement. The span attributes stuff does push the scope here, but if you're willing, I think this would be even better:

```python
expected_attributes = {
    'http.request.method': 'GET',
    'http.method': 'GET',
    'http.url': 'http://testserver/traced/',
    'http.scheme': 'http',
    'url.scheme': 'http',
    'http.status_code': 200,
    'http.response.status_code': 200,
    'server.port': 80,
    'client.address': '127.0.0.1',
    'network.protocol.version': '1.1',
}

if DJANGO_2_2:
    expected_attributes['http.route'] = '^traced/'

self.assertEqual(span.attributes, expected_attributes)
```

That's an example for `test_traced_get_both_semconv`. Then it would be clear whether or not `http.target` is in the span attributes. Right now if it is then the tests don't show it.

**Review by emdneto — APPROVED** [review]
Discussed in slack, prob we need to review other tests as well, but LGTM. Thanks for addressing the main comments.

File: instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py
Comment (lzchen) [reply instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py]: @alexmojaki 

This is a great point and we most likely need this scenario tested out.

I do not believe `http.target` was ever explicitly supported for `django` and only relies on the `wsgi` and `asgi` implementations of extracting this information. If `http.target` is missing, it simply would be a feature request of the django instrumentation and outside the scope of this pr. We also do not want to be including old semantic convention changes so the feature request would more accurately be "support url.path/url.query" in this case.

**Review by xrmx — APPROVED** [review]

**Comment by lzchen** [comment]
@alexmojaki 

Will wait on a review from you before merging!

**Review by alexmojaki — APPROVED** [review]

**Pull request merged**
